src/test/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureServiceSpec.scala

```diff
@@ -364,4 +364,29 @@ class AzureServiceSpec extends AnyFlatSpec with Matchers with ScalaFutures {
       mockMrgDAO.mrgs should not contain managedResourceGroup
     }
   }
+
+  "deleteManagedResourceGroup" should "delete a managed resource group" in {
+    val user = Generator.genWorkbenchUserAzure.sample
+    val managedResourceGroup = Generator.genManagedResourceGroup.sample.get
+    val mockMrgDAO = new MockAzureManagedResourceGroupDAO
+    val svc =
+      new AzureService(MockCrlService(user, managedResourceGroup.managedResourceGroupCoordinates.managedResourceGroupName), new MockDirectoryDAO(), mockMrgDAO)
+
+    mockMrgDAO.insertManagedResourceGroup(managedResourceGroup, samRequestContext).unsafeRunSync()
+    svc.deleteManagedResourceGroup(managedResourceGroup.billingProfileId, samRequestContext.copy(samUser = user)).unsafeRunSync()
+    mockMrgDAO.mrgs shouldNot contain(managedResourceGroup)
+  }
+
+  it should "NotFound when MRG does not exist" in {
+    val user = Generator.genWorkbenchUserAzure.sample
+    val mockMrgDAO = new MockAzureManagedResourceGroupDAO
+    val managedResourceGroup = Generator.genManagedResourceGroup.sample.get
+    val svc =
+      new AzureService(MockCrlService(user, managedResourceGroup.managedResourceGroupCoordinates.managedResourceGroupName), new MockDirectoryDAO(), mockMrgDAO)
+
+    val err = intercept[WorkbenchExceptionWithErrorReport] {
+      svc.deleteManagedResourceGroup(managedResourceGroup.billingProfileId, samRequestContext.copy(samUser = user)).unsafeRunSync()
+    }
+    err.errorReport.statusCode shouldBe Some(StatusCodes.NotFound)
+  }
```

Comment:
Are there any tests that make sense to add around auth or handling failure of auth?

Reply:
The assumption is that the assertion in the route for DELETE on the owning spend profile should be sufficient. The create methods in this service assert that the creating user is the one baked into the properties of the MRG -- I don't think we want to make a similar assertion at the time of deletion. That is, if I have DELETE permissions on the billing profile, the deletion should succeed regardless of the properties on the MRG.